### DISCUSSION TITLE
Traffic spikes reported that are bigger than the interfaces

### DISCUSSION THREAD

**Discussion by aurelienmzc** [akvorado/akvorado#572] [Q&A]
Hello,
I see traffic spikes reported that are bigger than the interfaces the traffic is traversing (25Gbps+ flows on 10Gbps ports etc)
Information exported from a mikrotik
A specific configuration is necessary?

**Comment by edgenative** [comment]
Following - I see the same thing on ROS7.8 and ROS7.9rc2 using IPFIX.  Setting the default sampling rate to 100 in Akvorado, and having the sampling interval of 1 and Sampling Space of 99 on the Mikrotik, it doesn't seem to sample correctly.  Seems to be fine, until there is a burst of traffic through the Mikrotik.

**Reply by chupztugz000000000** [reply]
@edgenative  did it fix for the graph? or in the bandwidth? just tweaking the sampling interval to 1 and sampling rate to 99?  or its a trial and testing?

**Reply by edgenative** [reply]
Nothing worked for me, I tried a good number of combinations in the end.  I ended up abandoning Mikrotik for this and a number of other reasons

**Reply by chupztugz000000000** [reply]
so your graph still incosistent? or still using it?

**Reply by edgenative** [reply]
I'm not using Mikrotik now.  With JunOS devices everything works fine. Been using this with Juniper now for a couple of years and all is great.

**Reply by chupztugz000000000** [reply]
Im using also JunOS, i have sampling rate of 100 and also in core 100, but i got spike like 5G for google and other interfaces. How did you do for you? Can you share your config? Or help me

**Reply by edgenative** [reply]
In JunOS I have my sampling rate set to 100 under forwarding-options and in akvorado I'm not specifying the sampling rate, since it can read what JunOS is sending.

I only have sampling enabled on my edge facing interfaces (e.g. peering, transit and customer)

Forwarding Options

forwarding-options {
sampling {
    instance {
        sample-ins {
            input {
                rate 100;
                max-packets-per-second 65535;
            }
            family inet {
                output {
                    flow-server x.x.x.x {
                        port 2055;
                        autonomous-system-type origin;
                        source-address x.x.x.x;
                        version9 {
                            template {
                                ipv4;
                            }
                        }
                    }
                    inline-jflow {
                        source-address x.x.x.x;
                    }
                }
            }
            family inet6 {
                output {
                    flow-server x.x.x.x {
                        port 2055;
                        autonomous-system-type origin;
                        source-address x.x.x.x;
                        version9 {
                            template {
                                ipv6;
                            }
                        }
                    }
                    inline-jflow {
                        source-address x.x.x.x;
                    }
                }
            }
        }
    }
}
}

Services

services {
    flow-monitoring {
        version9 {
            template ipv4 {
                flow-active-timeout 10;
                flow-inactive-timeout 10;
                template-refresh-rate {
                    packets 30;
                    seconds 30;
                }
                option-refresh-rate {
                    packets 30;
                    seconds 30;
                }
                ipv4-template;
            }
            template ipv6 {
                flow-active-timeout 10;
                flow-inactive-timeout 10;
                template-refresh-rate {
                    packets 30;
                    seconds 30;
                }
                option-refresh-rate {
                    packets 30;
                    seconds 30;
                }
                ipv6-template;
            }
        }
    }
}

**Reply by chupztugz000000000** [reply]
Thanks! l'll try this kind of setup!  did you alter any code also in the Akvorado side?  like in the inlet or outlet yaml file? i had just alter the default sampling rate to 100

**Reply by chupztugz000000000** [reply]
@edgenative  i have this in my inlet yaml flow:
  inputs:
    # NetFlow port
    - type: udp
      decoder: netflow
      listen: :9996
      workers: 6
      # Before increasing this value, look for it in the troubleshooting section
      # of the documentation.
      receive-buffer: 212992
    # IPFIX port
    - type: udp
      decoder: netflow
      listen: :4739
      workers: 6
      receive-buffer: 212992
    # sFlow port
    - type: udp
      decoder: sflow
      listen: :6343
      workers: 6
      receive-buffer: 212992

**Reply by edgenative** [reply]
flow:
    inputs:
      - type: udp
        decoder: netflow
        listen: 0.0.0.0:2055
        workers: 8
        receive-buffer: 10485760
      - type: udp
        decoder: sflow
        listen: 0.0.0.0:6343
        workers: 1
        receive-buffer: 10485760

**Reply by chupztugz000000000** [reply]
hi @edgenative thanks for helping me. now in my internal traffic all matches now. but theres a weird issue now for my external interface. i have put all the interface now. and only ge-0-0/3 the MPLS is the same graph other is not like my ISP and other external traffic

**Reply by chupztugz000000000** [reply]
Hi @edgenative for my internal traffic interface all are now matching. i have problem with my external interface like my ISP. it does not match . with my MPLS it match because its internal traffic. do you have any opinion on this? hope you can help me again

**Reply by edgenative** [reply]
What does the configuration of your outside interface look like?  Do you have your ASNs etc configured correctly in Akvorado?

**Reply by chupztugz000000000** [reply]
@edgenative  where can i look into it ? in the outlet.yaml? here is my  outlet.yaml `core:
  # Default sampling rate if exporter does not send it
  default-sampling-rate: 100
  workers: 6

  # Only use GeoIP (IPinfo) for ASN resolution
  asn-providers:
    - geo-ip

  # Classifiers for exporters (devices sending flows).
  exporter-classifiers:
    # Extract site from exporter name before the first '-'.
    # Example: "NYC-FW1" -> Site: "NYC".
    - ClassifySiteRegex(Exporter.Name, '^([^-]+)-', '$1')

    # Hardcode region, tenant, and role.
    - ClassifyRegion('')
    - ClassifyTenant('')
    - ClassifyRole('')

  # Classifiers for interfaces based on name and description.
  interface-classifiers:
    # Match external interfaces by name if they are ge- (Juniper gigabit ethernet)
    - |
      ClassifyConnectivityRegex(Interface.Name, '^(?i)ge-', 'ge') &&
      ClassifyExternal()

    # Match external interfaces such as Transit, PNI, PPNI, or IX from description
    - |
      ClassifyConnectivityRegex(Interface.Description, '^(?i)(transit|pni|ppni|ix):? ', '$1') &&
      ClassifyProviderRegex(Interface.Description, '^\\S+?\\s(\\S+)', '$1') &&
      ClassifyExternal()

    # Match point-to-point links (P2P) from description
    - |
      ClassifyConnectivityRegex(Interface.Description, '(?i)p2p', 'p2p') &&
      ClassifyExternal()

    # Match GRE interfaces by name
    - |
      ClassifyConnectivityRegex(Interface.Name, '^(?i)gr-', 'GRE') &&
      ClassifyExternal()

    # Default classification for everything else
    - ClassifyInternal()`

**Reply by chupztugz000000000** [reply]
my graph is the same now but the speed for TX for external or RX is not the same
